src/main/java/com/kamycz3q/coursesystemspringboot/personalData/controllers/PersonalDataCompanyController.java

```diff
@@ -0,0 +1,50 @@
+package com.kamycz3q.coursesystemspringboot.personalData.controllers;
+
+
+import com.kamycz3q.coursesystemspringboot.personalData.personalDataObject.PersonalDataCompany;
+import com.kamycz3q.coursesystemspringboot.personalData.record.CreatePersonalCompanyRequest;
+import com.kamycz3q.coursesystemspringboot.personalData.service.PersonalDataCompanyService;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/personal-data/company")
+public class PersonalDataCompanyController {
```

Comment:
`PersonalData` i `PersonalCompanyData`. To drugie nie ma sensu moim zdaniem, bo `PersonalData` oznacza dane osobowe. Dane osobowe mogą dotyczyć tylko osób fizycznych. Dla osób prawnych (firm) powinno istnieć raczej `CompanyData`. I teraz założenie biznesowe jest takie, że większość klientów (Customer), będzie posiadać firmę i poda jej dane. Ale będą też przypadki takie, że klientem jest osoba fizyczna i tej firmy nie posiada.

Reply:
Czyli byt Customer powinien agregować PersonalData oraz CompanyData, gdzie to drugie może być puste (ponieważ nie dotyczy tego klienta). Ponieważ Customer jest agregatem (agreguje PersonalData i CompanyData), to top-level package w javie powinien być com.kamycz3q.coursesystemspringboot.customer. W środku powinien być subpackage .model, a w nim klasy Customer, PersonalData i CompanyData. Taki sposób na organizację danych w klaach modelowych, gdzie nie korzystamy z dziedziczenia klas (słówko kluczowe `extends` w javie) nazywa się kompozycja. Ponieważ komponujemy klasę z innych klas. Generalnie dobrą praktyką jest korzystać jak najmniej z dziedziczenia, a jak najwięcej z kompozycji, ponieważ kiedy chcemy coś zmienić w strukturze danych, to łatwiej jest zdekomponować klasę i skomponować na nowo, niż odczarowywać ciąg dziedziczenia (może być 5 poziomów dziedziczenia i wtedy jest dupa).